src/fungibled/runtime.rs

```diff
@@ -235,18 +236,65 @@ impl Runtime {
         //       of asset for the transfer operation
 
         trace!("Looking for asset information");
-        let mut asset = self.cacher.asset(transfer.contract_id)?.clone();
-        debug!("Transferring asset {}", asset);
+        debug!("Transferring asset {}", transfer.contract_id);
 
         trace!("Preparing state transition");
         let transition = rgb20::transfer(
-            &mut asset,
+            self.cacher.asset(transfer.contract_id)?,
             transfer.inputs.clone(),
             transfer.payment.clone(),
             transfer.change.clone(),
         )?;
         debug!("State transition: {}", transition);
 
+        trace!("Collecting other assets on the spent outpoints and preparing blank state transitions");
+        let mut other_outpoint_assets: BTreeMap<
+            ContractId,
+            BTreeSet<(OutPoint, AtomicValue)>,
+        > = bmap! {};
+        for outpoint in &transfer.inputs {
+            for (other_contract, amounts) in
+                self.cacher.outpoint_assets(*outpoint)?
+            {
+                other_outpoint_assets
+                    .entry(other_contract)
+                    .or_insert(empty!())
```

Comment:
Shouldn't we `other_contract` here? Why we are keeping it `empty!()` if we dont find a contract entry?

Reply:
Because the actual insert operation happens one line below; here we initi the empty structure if it was not in place (amd inserting data from other contracts in the next line). It's the way `BTreeMap::entry` is working.